crates/topos-p2p/src/runtime/mod.rs

```diff
@@ -62,18 +61,19 @@ impl Runtime {
 
         self.bootstrapped = true;
 
-        debug!("Added external address: {:?}", self.addresses);
-        self.swarm.add_external_address(self.addresses.clone());
+        debug!("Added external addresses: {:?}", self.advertised_addresses);
+        for address in &self.advertised_addresses {
+            self.swarm.add_external_address(address.clone());
+        }
 
         debug!("Starting to listen on {:?}", self.listening_on);
-        let addr = self.listening_on.clone();
-        if let Err(error) = self.swarm.listen_on(addr) {
-            error!(
-                "Couldn't start listening on {} because of {error:?}",
-                self.listening_on
-            );
+        let addresses = self.listening_on.clone();
+        for addr in addresses {
+            if let Err(error) = self.swarm.listen_on(addr.clone()) {
+                error!("Couldn't start listening on {} because of {error:?}", addr);
```

Comment:
```suggestion
        for addr in self.listening_on {
            if let Err(error) = self.swarm.listen_on(addr.clone()) {
                error!("Couldn't start listening on {} because of {error:?}", addr);
```